main.py

```diff
@@ -106,10 +106,47 @@ def get_name_tag(resource_type, name):
     ]
 )
 
+# Create EC2 for public subnets
+instances = []
+for i, public_subnet in enumerate(public_subnets):
+    instance = ec2.create_instances(
+        ImageId='ami-0c0b74d29acd0cd97',
```

Comment:
We shouldn't hard-code the AMI IDs, since they might differ by regions. For example, in my region the code throws an exception because of this.

We could however do someting like this to get x86_64 Amazon Linux 2023 AMI dynamically:

```
client = boto3.client('ec2')
ami_filters = [
    {
        'Name': 'name',
        'Values': [
            'Amazon Linux 2023 AMI 2023.0.20230315.0 x86_64 HVM kernel-6.1 SSD Volume Type by Venv'
        ]
    },
    {
        'Name': 'architecture',
        'Values': [
            'x86_64'
        ]
    }
]
amis = client.describe_images(Filters=ami_filters)
selected_ami = amis['Images'][0]['ImageId']
```

Then we can use the selected_ami using its ImageId:

```         ImageId=selected_ami,```

Reply:
Feel free to make adjustements to make it nicer if you have any ideas.